app/src/main/java/org/mozilla/fenix/home/pocket/PocketCategoriesViewHolder.kt

```diff
@@ -0,0 +1,111 @@
+
+@file:Suppress("MagicNumber")
+
+package org.mozilla.fenix.home.pocket
+
+import android.view.View
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.SectionHeader
+import org.mozilla.fenix.home.HomeFragmentStore
+import org.mozilla.fenix.home.topsites.dpToPx
+import org.mozilla.fenix.theme.FirefoxTheme
+
+internal const val POCKET_CATEGORIES_SELECTED_AT_A_TIME_COUNT = 8
+
+/**
+ * [RecyclerView.ViewHolder] for displaying the list of [PocketRecommendedStoriesCategory]s from [HomeFragmentStore].
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param store [HomeFragmentStore] containing the list of Pocket stories categories to be displayed.
+ * @param interactor [PocketStoriesInteractor] callback for user interaction.
+ */
+class PocketCategoriesViewHolder(
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val store: HomeFragmentStore,
+    private val interactor: PocketStoriesInteractor
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+
+    @Composable
+    override fun Content() {
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        val topPadding = composeView.dpToPx(24f)
+        composeView.setPadding(horizontalPadding, topPadding, horizontalPadding, 0)
+
+        val categories = store
+            .observeAsComposableState { state -> state.pocketStoriesCategories }.value
+        val categoriesSelections = store
+            .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
+
+        PocketTopics(
+            categories = categories,
+            categoriesSelections = categoriesSelections,
+            onCategoryClick = interactor::onCategoryClicked
+        )
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+    }
+}
+
+@Composable
+private fun PocketTopics(
+    categories: List<PocketRecommendedStoriesCategory>?,
+    categoriesSelections: List<PocketRecommendedStoriesSelectedCategory>?,
```

Comment:
Can we change the signature of this so that the `categories` and `categoriesSelections` are not nullable, and we can just pass in the `emptyList()` in `Content()`